### PR TITLE
Resolve #523, #1230, #1319

### LINKED ISSUE #523: [Test] LandingPage
1. test for all btns in nav drawer, check for correct redirection link
2. test for nav drawer footer
    - contains copy right
    - help link
    - terms link
    - build version number exists
3. look for "Email Transfer" title
4. look for subtitle "Description goes here..."
5. Check "Start Email Transfer" btn's link
6. Check "Connect Your Accounts btn's link
7. Look for "Recent Transactions" title
8. Look for empty transaction placeholder img
9. Look for youtube embed

### PR DESCRIPTION
**Pull request by InVincible2016**

### CODE REVIEW

File: src/tests/e2e/login.test.e2e.js [hunk 1]
```diff
@@ -18,4 +27,43 @@ describe('Login and onboarding', () => {
...
+      // close non in
+      await page.close()
+      // first login to disconnect page to delete cloud wallet
+      let context = await browser.createIncognitoBrowserContext()
+      // Create a new page inside context.
+      page = await context.newPage()
+
+      await page.goto(`${process.env.E2E_TEST_URL}/disconnect`)
+      const loginPage = new LoginPage()
+      await loginPage.login(
```
Comment (timothywangdev) [inline src/tests/e2e/login.test.e2e.js]: aren't we at disconnect page, how do we login?

File: src/tests/e2e/login.test.e2e.js [hunk 2]
```diff
@@ -18,4 +27,43 @@ describe('Login and onboarding', () => {
...
+      await page.goto(`${process.env.E2E_TEST_URL}`)
+      // now test account is a new user
+      await loginPage.login(
+        process.env.E2E_TEST_GOOGLE_LOGIN_USERNAME,
+        process.env.E2E_TEST_GOOGLE_LOGIN_PASSWORD
+      )
+
+      // once fully login, check if there is any error snakbar
+      // wait for 5 seconds
+      await sleep(5000)
```
Comment (timothywangdev) [inline src/tests/e2e/login.test.e2e.js]: no need to sleep, loginPage.login already waits till all network calls finish

File: src/tests/e2e/pages/login.page.js [hunk 3]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
-          pageCount = (await browser.pages()).length
-        })()
-        return pageCount
-      }, 3)
-      log.info('Popup opened')
-      var browserPages = await browser.pages()
-      var googleLoginPopup = browserPages.reduce(function (acc, curr) {
-        if (curr.url().indexOf('google') !== -1) {
-          return curr
+    var pageCount = 0
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: remove this if not needed

File: src/tests/e2e/pages/login.page.js [hunk 4]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
-            log.error('Unable to find captcha solution')
-          }
-        }
-      }
+    // default to treat every login as a new user
+    let newUser = true
+    try {
+      await sleep(3000)
+      // new user would get a authorization page
+      await expect(googleLoginPopup).toClick('span', { text: 'Allow', timeout: 3000 })
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: i believe this requires actual drive disconnect to reset the permissions

File: src/tests/e2e/pages/login.page.js [hunk 5]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
 class LoginPage {
   async login (username, password, mockLogin = false) {
     log.debug(username, password)
-    // assume we are at login page
-
+    
+    let currentPageCount = (await browser.pages()).length
     // close cookie consent
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: wrong position for the comment

File: src/tests/e2e/pages/login.page.js [hunk 6]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
 class LoginPage {
   async login (username, password, mockLogin = false) {
     log.debug(username, password)
-    // assume we are at login page
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: why remove this?

File: src/tests/e2e/pages/login.page.js [hunk 7]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
-        })()
-        return pageCount
-      }, 3)
-      log.info('Popup opened')
-      var browserPages = await browser.pages()
-      var googleLoginPopup = browserPages.reduce(function (acc, curr) {
-        if (curr.url().indexOf('google') !== -1) {
-          return curr
+    var pageCount = 0
+    await runUntilEvaluateEquals(function () {
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: if {} indentation problems

File: src/tests/e2e/pages/login.page.js [hunk 8]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
 class LoginPage {
   async login (username, password, mockLogin = false) {
     log.debug(username, password)
-    // assume we are at login page
-
+    
+    let currentPageCount = (await browser.pages()).length
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: move inside if(!mocklogin){...}, this is only relevant if mock user is not used

File: src/tests/e2e/pages/login.page.js [hunk 9]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
+    } catch (e) {
+      // this is not a new user if authorization screen is not displayed
+      newUser = false
+    }
 
-      await googleLoginPopup.waitForFunction('document.querySelector("input")')
-      log.info('Google login popup password field loaded')
-      await sleep(1000)
+    await sleep(2000)
+    await expect(page).toMatchElement('p', { text: 'Loading...', timeout: 100000 })
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: Reduce timeout to something reasonable rather than 100s

File: src/tests/e2e/pages/login.page.js [hunk 10]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
 
-      await googleLoginPopup.keyboard.type(password)
-      await googleLoginPopup.keyboard.press('Enter')
+    if (newUser) {
+      // new user would get a authorization page
+      await expect(page).toMatchElement('p', {
+        text: 'Setting up your account...',
+        timeout: 100000
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: Reduce timeout to something reasonable rather than 100s

File: src/tests/e2e/pages/login.page.js [hunk 11]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
+      })
+      await expect(page).toMatchElement('p', { text: "It won't take long.", timeout: 100000 })
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: Reduce timeout to something reasonable rather than 100s

File: src/tests/e2e/pages/login.page.js [hunk 12]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
+      log.info('setting up chainsfr wallet accounts displayed')
+      await sleep(5000)
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: no need to sleep here

File: src/tests/e2e/pages/login.page.js [hunk 13]
```diff
@@ -29,71 +29,95 @@ const sleep = milliseconds => {
...
+      await expect(googleLoginPopup).toClick('span', { text: 'Allow', timeout: 3000 })
+    } catch (e) {
+      // this is not a new user if authorization screen is not displayed
+      newUser = false
+    }
 
-      await googleLoginPopup.waitForFunction('document.querySelector("input")')
-      log.info('Google login popup password field loaded')
-      await sleep(1000)
+    await sleep(2000)
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: this sleep may cause line 186 to fail if loading finishes

File: src/components/LandingPageComponent.jsx [hunk 14]
```diff
@@ -551,15 +552,16 @@ class LandingPageComponent extends Component {
                 height='100%'
                 justifyContent='center'
               >
-                <Typography variant='h2'>Email Transfer</Typography>
-                <Typography className={classes.descText}>Description goes here...</Typography>
+                <Typography variant='h2' data-test-id='emtTitle'>Email Transfer</Typography>
+                <Typography className={classes.descText} data-test-id='emtSubtitle'>Description goes here...</Typography>
```
Comment (timothywangdev) [inline src/components/LandingPageComponent.jsx]: line too long

File: src/tests/e2e/landing.test.e2e.js [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+const timeout = 180000
+
+describe('Landing page tests', () => {
+  beforeAll(async () => {
+    await page.goto(process.env.E2E_TEST_URL)
+    // login to app
+    const loginPage = new LoginPage()
+    await loginPage.login(
+      process.env.E2E_TEST_GOOGLE_LOGIN_USERNAME,
+      process.env.E2E_TEST_GOOGLE_LOGIN_PASSWORD
```
Comment (timothywangdev) [inline src/tests/e2e/landing.test.e2e.js]: use mock user for landing test

File: src/tests/e2e/landing.test.e2e.js [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+    )
+  }, timeout)
+
+  afterAll(async () => {
+    await jestPuppeteer.resetBrowser()
```
Comment (timothywangdev) [inline src/tests/e2e/landing.test.e2e.js]: do we need to reset if using incognito?

File: src/tests/e2e/landing.test.e2e.js [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+  })
+
+  afterEach(async () => {
+    await page.goto(process.env.E2E_TEST_URL)
+  })
+
+  it(
+    'navigate to email transfer',
```
Comment (timothywangdev) [inline src/tests/e2e/landing.test.e2e.js]: first letter capital

File: src/tests/e2e/landing.test.e2e.js [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+      await landingPage.clickHelpCenter()
+      await landingPage.clickTerms()
+      await landingPage.checkCopyRight()
+      await landingPage.checkBuild()
+    },
+    timeout
+  )
+
+  it(
+    'main page texts',
```
Comment (timothywangdev) [inline src/tests/e2e/landing.test.e2e.js]: first letter capital

Comment (timothywangdev) [reply src/tests/e2e/landing.test.e2e.js]: "Landing page static elements"

File: src/tests/e2e/landing.test.e2e.js [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+  }, timeout)
+
+  afterAll(async () => {
+    await jestPuppeteer.resetBrowser()
+  })
+
+  afterEach(async () => {
+    await page.goto(process.env.E2E_TEST_URL)
+  })
+
```
Comment (timothywangdev) [inline src/tests/e2e/landing.test.e2e.js]: missing 6. Check "Connect Your Accounts" btn's link

File: src/tests/e2e/login.test.e2e.js [hunk 20]
```diff
@@ -49,7 +47,7 @@ describe('Login and onboarding', () => {
 
       // reset page and close incognito page
       await jestPuppeteer.resetPage()
-        
+
       context = await browser.createIncognitoBrowserContext()
```
Comment (timothywangdev) [inline src/tests/e2e/login.test.e2e.js]: already use incognito by default

File: src/tests/e2e/login.test.e2e.js [hunk 21]
```diff
@@ -1,20 +1,18 @@
 import LoginPage from './pages/login.page'
```
Comment (timothywangdev) [inline src/tests/e2e/login.test.e2e.js]: this file should be in previous commit

File: src/tests/e2e/pages/login.page.js [hunk 22]
```diff
@@ -1,31 +1,10 @@
 import log from 'loglevel'
 import captcha from 'async-captcha'
```
Comment (timothywangdev) [inline src/tests/e2e/pages/login.page.js]: move this changes into previous commit

File: src/tests/e2e/testUtils.js [hunk 23]
```diff
@@ -0,0 +1,45 @@
+const sleep = milliseconds => {
```
Comment (timothywangdev) [inline src/tests/e2e/testUtils.js]: move this changes into previous commit

File: src/tests/e2e/pages/login.page.js (hunk 4 above)
Comment (timothywangdev) [reply src/tests/e2e/pages/login.page.js]: disconnect revokes the scope

File: src/tests/e2e/landing.test.e2e.js (hunk 19 above)
Comment (InVincible2016) [reply src/tests/e2e/landing.test.e2e.js]: it's tested

File: src/components/LandingPageComponent.jsx (hunk 14 above)
Comment (InVincible2016) [reply src/components/LandingPageComponent.jsx]: fixed

File: src/tests/e2e/landing.test.e2e.js (hunk 16 above)
Comment (InVincible2016) [reply src/tests/e2e/landing.test.e2e.js]: yes, to ensure this test would not interfere with other tests.

File: src/tests/e2e/landing.test.e2e.js (hunk 17 above)
Comment (InVincible2016) [reply src/tests/e2e/landing.test.e2e.js]: fixed

File: src/tests/e2e/landing.test.e2e.js (hunk 18 above)
Comment (InVincible2016) [reply src/tests/e2e/landing.test.e2e.js]: fixed

File: src/tests/e2e/login.test.e2e.js [hunk 24]
```diff
@@ -18,4 +27,43 @@ describe('Login and onboarding', () => {
     },
     timeout
   )
+
+  it(
+    'Login as a new user',
+    async () => {
+      // close non in
```
[the remark this answers is not in the document]
Comment (InVincible2016) [reply src/tests/e2e/login.test.e2e.js]: fixed

File: src/tests/e2e/login.test.e2e.js [hunk 25]
```diff
@@ -18,4 +27,43 @@ describe('Login and onboarding', () => {
...
+      await page.close()
+      // first login to disconnect page to delete cloud wallet
+      let context = await browser.createIncognitoBrowserContext()
```
[the remark this answers is not in the document]
Comment (InVincible2016) [reply src/tests/e2e/login.test.e2e.js]: fixed

File: src/tests/e2e/login.test.e2e.js (hunk 1 above)
Comment (InVincible2016) [reply src/tests/e2e/login.test.e2e.js]: the '/disconnct' requires login first

File: src/tests/e2e/login.test.e2e.js (hunk 20 above)
Comment (InVincible2016) [reply src/tests/e2e/login.test.e2e.js]: fixed

File: src/tests/e2e/login.test.e2e.js (hunk 2 above)
Comment (InVincible2016) [reply src/tests/e2e/login.test.e2e.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 8 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 3 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 7 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 13 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 9 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 10 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 11 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

File: src/tests/e2e/pages/login.page.js (hunk 12 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/login.page.js]: fixed

**Pull request merged**